i18n_generate.mll

```diff
@@ -118,50 +132,87 @@ let print_header fmt ?primary_module ~default_language () =
   in
   Format.pp_print_string fmt @@
   default_lang ^
-   "let%server _language_ = " ^ server_language_reference ^ "\n\
-   let%server get_language () = Eliom_reference.Volatile.get _language_\n\
-   let%server set_language language = \n\
-   Eliom_reference.Volatile.set _language_ language\n\
-   \n\
-   let%client _language_ = " ^ client_language_reference ^ "\n\
-   let%client get_language () = !_language_\n\
-   let%client set_language language = _language_ := language\n\
+  "let%server _language_ = " ^ server_language_reference ^ "\n\
+  let%server get_language () = Eliom_reference.Volatile.get _language_\n\
+  let%server set_language language = \n\
+  Eliom_reference.Volatile.set _language_ language\n\
+  \n\
+  let%client _language_ = " ^ client_language_reference ^ "\n\
+  let%client get_language () = !_language_\n\
+  let%client set_language language = _language_ := language\n\
+  \n\
+  let%shared txt = Eliom_content.Html.F.txt\n\
+  "
+
+let print_header fmt ?primary_module ~default_language () =
+  let default_lang =
+    match primary_module with
+    | None -> "let default_language = " ^ default_language ^ "\n"
+    | Some module_name -> ""
+  in
+  Format.pp_print_string fmt @@
+  default_lang ^
+  "let _language_ = default_language \n\
+   let get_language () = your_function_to_getting_language \n\
+   let set_language language = \n\
+   your_function_to_setting_language\n\
    \n\
-   let%shared txt = Eliom_content.Html.F.txt\n\
-"
+   let txt = txt \n\
```

Comment:
is this needed?

Reply:
It's needed because Eliom_content is a part of Eliom project : http://ocsigen.org/eliom/latest/api/server/Eliom_content.Html.F
And we want to remove the eliom dependance